### PR TITLE
Fix config sections is no longer an associative array

### PR DESCRIPTION
**Pull request by flamerohr**
This is so that order priority is enforced when json.stringify is used.

Pull requests which rely on this:
https://github.com/silverstripe/silverstripe-asset-admin/pull/435
https://github.com/silverstripe/silverstripe-campaign-admin/pull/10
https://github.com/silverstripe/silverstripe-frameworktest/pull/33

### CODE REVIEW

File: code/LeftAndMain.php [hunk 1]
```diff
@@ -306,6 +306,7 @@ public function getClientConfig()
         return [
             // Trim leading/trailing slash to make it easier to concatenate URL
             // and use in routing definitions.
+            'name' => $this->class,
```
Comment (tractorcow) [inline code/LeftAndMain.php]: `$this->class` is going soon as we remove Object class.

Comment (flamerohr) [inline code/LeftAndMain.php]: good point, I've changed to `static:class` :D

**Comment by tractorcow** [comment]
Looks good, will need a rebase soon. :)

**Comment by tractorcow** [comment]
Tests are broken and I think the comments need updating in BootRouter.js. Can you please fix @flamerohr ?

**Pull request merged**